packages/roosterjs-plugin-picker/lib/PickerPlugin.ts

```diff
@@ -123,6 +124,18 @@ export default class PickerPlugin implements EditorPickerPluginInterface {
      * @param event PluginEvent object
      */
     public onPluginEvent(event: PluginEvent) {
+        if (event.eventType == PluginEventType.ContentChanged &&
+            event.source == ChangeSource.SetContent && this.dataProvider.onContentChanged) {
+                // Undos and other major changes to document content fire this type of event.
+                // Inform the data provider of the current picker placed elements in the body.
+                const elementsInDocument = this.editor.getDocument()
+                                            .querySelectorAll("[id^='" + this.pickerOptions.elementIdPrefix + "']");
+                let elementIds: string[] = [];
+                elementsInDocument.forEach(element => {

```

Comment:
and editor.queryElements() has a forEachCallback parameter, you can use it to do the loop.